format/udf-spec.md

```diff
@@ -0,0 +1,402 @@
+   - Names and their relative ordering are immutable. Only appending new names is allowed.
+   - Only the `doc` field may be updated in place.
+   - Overloads that differ only by parameter types are allowed (e.g., `foo(int)`, `foo(float)`, `foo(string, string)`), and
+     they all derive their parameter names positionally from the same `parameter-names` prefix.
+
+### Definition
+
+Each `definition` represents one function signature (e.g., `add_one(int)` vs `add_one(float)`).
+
+| Requirement | Field name           | Type                                                                                                                                                                                                                                                        | Description                                                                                                                                                                                         |
+|-------------|----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| *required*  | `definition-id`      | `string`                                                                                                                                                                                                                                                    | An identifier derived from canonical parameter-type tuple (lowercase, no spaces; e.g., `"(int,int,string)"`). If longer than 128 chars, use hashed form `"sig1-<base32(SHA-256(signature))[:26]>"`. |
+| *required*  | `parameters`         | `list<parameter>`                                                                                                                                                                                                                                           | Ordered list of [function parameters](#parameter). Invocation order **must** match this list.                                                                                                       |
+| *required*  | `return-type`        | [Iceberg Type JSON Representation](https://iceberg.apache.org/spec/#appendix-c-json-serialization), primitive types are encoded as JSON strings (e.g., `"int"`, `"string"`), and complex types are encoded as JSON objects (e.g., `{"type": "struct", ...}` | Type of value returned                                                                                                                                                                              |
```

Comment:
Tables are really hard to work with when they have embedded URLs like this. I think a best practice is to use a named URL:

```
[text][link-name]
...
[link-name]: https://...
```

I also highly recommend keeping table content short and minimal. If you need more clarification for how a field is used or interpreted, the additional information should go in a paragraph or number list of notes after the table.

Reply:
I will reword a bit.